build/config/siso/brave_siso_config.star

```diff
@@ -0,0 +1,278 @@
+# -*- bazel-starlark -*-
+# Copyright (c) 2025 The Brave Authors. All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at https://mozilla.org/MPL/2.0/.
+"""Siso configuration adjustments for Brave browser."""
+
+load("@builtin//struct.star", "module", "struct")
+
+# Rules to disable remote execution.
+__RULES_TO_DISABLE_REMOTE = [
+    # We have chromium_src mangling which requires additional inputs/outputs
+    # handling. This is not trivial and require careful configuration. Can be
+    # revisited when SISO becomes first class citizen in Brave.
+    "typescript/ts_library",
+]
+
+# Labels to remove from platforms (EngFlow-specific).
+__LABELS_TO_REMOVE_FROM_PLATFORMS = [
+    "action_default",
+    "action_large",
+]
+
+# Chromium_src overrides we support with the exception of c/c++/obj-c files
+# which are handled by redirect_cc. The list should be in sync with
+# `touchOverriddenFiles()` in //brave/build/commands/lib/util.js .
+__BRAVE_CHROMIUM_SRC_FILE_EXTENSIONS = (
+    '.css',
+    '.html',
+    '.icon',
+    '.json',
+    '.mojom',
+    '.pdl',
+    '.py',
+    '.ts',
+    '.xml',
+)
+
+# A python remote wrapper to add PYTHONPATH to the environment. Generated during
+# runhooks by //brave/third_party/reclient_configs/brave_custom/brave_custom.py.
+# The path should be relative to working directory which is `out/<config>`.
+__PYTHON_REMOTE_WRAPPER = "../../buildtools/reclient_cfgs/python/python_remote_wrapper"
+
+# Set to True to enable debug logging for handlers in this file.
+__debug = False
+
+
+# Main configuration function that sets up SISO for Brave-specific build
+# requirements by adjusting step configurations and registering custom handlers.
+def __configure(ctx, step_config, handlers):
+    __adjust_handlers(ctx, step_config, handlers)
+    __remove_labels_from_platforms(step_config)
+
+
+# Configures handlers to handle chromium_src overrides and disable remote
+# execution for specific rules that require local processing.
+def __adjust_handlers(ctx, step_config, handlers):
+    # Register the default redirect_cc handler.
+    handlers["redirect_cc"] = __redirect_cc_handler
+
+    # Adjust rules.
+    for rule in step_config["rules"]:
+        rule_name = rule["name"]
+
+        # Disable remote execution for rules that need local processing
+        if rule_name in __RULES_TO_DISABLE_REMOTE:
```

Comment:
There are two remote features:
- remote cache
- remote execution

Does it disable both or only the latter?

Reply:
both I believe. There's no any mention of keeping the cache or only disabling the remote execution in `remote` flag description https://chromium.googlesource.com/infra/infra/+/refs/heads/main/go/src/infra/build/siso/docs/starlark_config.md#initialization

Reply:
asked upstream, lets see